### PR TITLE
Add support for type aliases

### LINKED ISSUE #1544: Type aliases
### Relevant Discussion

https://github.com/pyccel/pyccel/issues/1487#issuecomment-1702207633

### Describe the feature
As suggested by @adam-urbanczyk .

It would be nice to have support for type aliases. This would allow the user to change the types of multiple functions at the same time. E.g. it is common to use this in C++ with a type alias `using Real = double;`

### Test Code
Provide code which does not currently work but which should do when this issue is fixed:
```python
from typing import TypeAlias

MyType: TypeAlias = int
```

### Proposed Solution
This could easily be implemented after #1523 by saving the type alias as a symbolic alias in the scope. An error should be raised if the type cannot be determined at compile time (e.g. because it is defined in an if block).

In order to implement this the following steps should be taken:
1. In `pyccel/ast/typingext.py` create a class inheriting from `pyccel.ast.datatypes.DataType` which describes the type `TypeAlias`
2. Add the module to the `pyccel.ast.utilities.builtin_import_registry`
3. In the semantic stage, make the necessary changes in `_visit_Assign` such that if the `lhs` is a `TypeAlias`, the alias is added to the scope using `pyccel.parser.scope.Scope.insert_symbolic_alias` and an `EmptyNode()` is returned
4. Add tests

### PR DESCRIPTION
**Pull request by EmilyBourne**
Add support for non-parametrised type aliases as described at https://docs.python.org/3/library/typing.html#type-aliases. Fixes #1544

**Commit Summary**
- Add a `TypeAlias` datatype
- Add support for `typing.TypeAlias`
- Ensure errors are raised when symbolic objects are redefined
- Avoid `class_type.name` (only available for types of user-defined classes)
- Remove dead code "handling lambdify"
- Stop calling `insert_symbolic_function` (unused, see #330 )
- Add tests

### CODE REVIEW

File: tests/errors/semantic/non_blocking/wrong_class_type_annotation.py [hunk 1]
```diff
@@ -11,4 +11,3 @@ def get_4(self : 'A'):
 if __name__ == '__main__':
     myA : int = A()
 
-    print(myA.x)
```
Comment (EmilyBourne) [inline tests/errors/semantic/non_blocking/wrong_class_type_annotation.py:14]: `myA` now initialised as `int` so `x` not found leading to blocking error. This line is removed as it is not what we want to test in this file

**Comment by EmilyBourne** [comment]
Failing tests are passing on Python 3.12

**Review by mustapha-belbiad — APPROVED** [review]
Looks Good to me!

**Review by jalalium — APPROVED** [review]

File: pyccel/ast/class_defs.py [hunk 2]
```diff
@@ -265,6 +265,8 @@ def get_cls_base(class_type):
         return SetClass
     elif isinstance(class_type, DictType):
         return DictClass
+    elif isinstance(class_type, SymbolicType):
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/class_defs.py:268]: I agree. Fixed : 8242e419362b1809cec3530aa79b6752e469f315

[the remark this answers is not in the document]
Comment (jalalium) [reply pyccel/ast/class_defs.py]: Thank you!

**Pull request merged**